rust/sedona-spatial-join/src/build_index.rs

```diff
@@ -33,7 +33,72 @@ use crate::{
     spatial_predicate::SpatialPredicate,
 };
 
-pub(crate) async fn build_index(
+/// Synchronous version of build_index that doesn't spawn tasks
+/// Used in execution contexts without async runtime support (e.g., Spark/Comet JNI)
+pub async fn build_index_sync(
```

Comment:
This newly added `build_index_sync` is also an async function. Why can we not directly use `build_index`, which is also an async function?

For the use case of Comet, I believe that the DataFusion physical plans constructed by Comet are all single partition. What problem have you encountered when using `build_index`?